ethcore/types/src/transaction/transaction.rs

```diff
@@ -152,59 +212,134 @@ impl HeapSizeOf for Transaction {
     }
 }
 
-#[cfg(any(test, feature = "test-helpers"))]
-impl From<ethjson::state::Transaction> for SignedTransaction {
-    fn from(t: ethjson::state::Transaction) -> Self {
-        let to: Option<ethjson::hash::Address> = t.to.into();
-        let secret = t.secret.map(|s| Secret::from(s.0));
-        let tx = Transaction {
-            nonce: t.nonce.into(),
-            gas_price: t.gas_price.into(),
-            gas: t.gas_limit.into(),
-            action: match to {
-                Some(to) => Action::Call(to.into()),
-                None => Action::Create,
-            },
-            value: t.value.into(),
-            data: t.data.into(),
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct AccessListTx {
+    pub transaction: Transaction,
+    //optional access list
+    pub access_list: Vec<(H160, Vec<H256>)>,
+}
+
+impl AccessListTx {
+    pub fn new(transaction: Transaction, access_list: Vec<(H160, Vec<H256>)>) -> AccessListTx {
+        AccessListTx {
+            transaction,
+            access_list,
+        }
+    }
+
+    pub fn tx(&self) -> &Transaction {
+        &self.transaction
+    }
+
+    pub fn tx_mut(&mut self) -> &mut Transaction {
+        &mut self.transaction
+    }
+
+    // decode bytes by this payload spec: rlp([3, [nonce, gasPrice, gasLimit, to, value, data, access_list, senderV, senderR, senderS]])
+    pub fn decode(tx: &[u8]) -> Result<UnverifiedTransaction, DecoderError> {
+        let tx_rlp = &Rlp::new(&tx[1..]); //first byte is related to transaction type defined in EIP-2718
+
+        // we need to have 10 items in this list
+        if tx_rlp.item_count()? != 10 {
+            return Err(DecoderError::RlpIncorrectListLen);
+        }
+        // first part of list is same as legacy transaction and we are reusing that part.
+        let transaction = Transaction::decode_data(&tx_rlp)?;
+
+        // access list we get from here
+        let accl_rlp = tx_rlp.at(6)?;
+
+        // access_list pattern: [[{20 bytes}, [{32 bytes}...]]...]
+        let mut accl: Vec<(H160, Vec<H256>)> = Vec::new();
+
+        for i in 0..accl_rlp.item_count()? {
+            let accounts = accl_rlp.at(i)?;
+
+            if accounts.item_count()? != 2 {
+                //TODO check what to do if we have only one item, should we be strict or not
```

Comment:
I think that this is ok, in EIP2929 `process_access_list` pseudocode says

`assert isinstance(item, list) and len(item) == 2`

Reply:
Agreed, removing the comment